src/app-layer-protos.c

```diff
@@ -97,6 +99,8 @@ void AppProtoRegisterProtoString(AppProto alproto, const char *proto_name)
             g_alproto_strings = tmp;
         }
         g_alproto_max++;
+        SCAppLayerParserReallocCtx(alproto);
+        SCAppLayerProtoDetectReallocCtx(alproto);
```

Comment:
I do not think we need `SCAppLayerProtoDetectReallocCtx` here for detection-only, as it gets grown with the current code already when we register detection, right ?